lib/facter/util/facts/windows_release_finder.rb

```diff
@@ -26,6 +26,8 @@ def check_version_10(consumerrel, kernel_version)
             build_number = kernel_version[/([^.]*)$/].to_i
             if consumerrel
```

Comment:
I think we should decompose this `if` sequence to specific `if` by variable and with easy to read limits/values